Lib/asyncio/selector_events.py

```diff
@@ -789,6 +788,11 @@ def resume_reading(self):
         if self._loop.get_debug():
             logger.debug("%r resumes reading", self)
 
+    def _add_reader(self, fd, callback, *args):
```

Comment:
`_SelectorTransport` has `_add_reader()` method already.
I think the former should be modified to check `is_reading()` instead of just `self._closing`, `_SelectorSocketTrasport._add_reader()` should be removed.

Is there any reason to not do it?

Reply:
I think you're right.
When I started changing it, I asked my self why ```pause_reading``` and ```resume_reading```  are not implemented in ```_SelectorTransport```, so I changed it too.
Is there any reason why it will be a bad idea?